math/mathcore/inc/Math/CladDerivator.h

```diff
@@ -19,420 +19,221 @@
+}
+
+template <typename T>
+ValueAndPushforward<Double_t, Double_t> ATan_pushforward(T x, T d_x)
+{
+   return {::TMath::ATan(x), (1. / (x * x + 1)) * d_x};
+}
+
+template <typename T>
+ValueAndPushforward<Double_t, Double_t> ATanH_pushforward(T x, T d_x)
+{
+   return {::TMath::ATanH(x), (1. / (1 - x * x)) * d_x};
+}
+
+template <typename T>
+ValueAndPushforward<T, T> Cos_pushforward(T x, T d_x)
+{
+   return {::TMath::Cos(x), -::TMath::Sin(x) * d_x};
+}
+
+template <typename T>
+ValueAndPushforward<T, T> CosH_pushforward(T x, T d_x)
+{
+   return {::TMath::CosH(x), ::TMath::SinH(x) * d_x};
+}
+
+template <typename T>
+ValueAndPushforward<Double_t, Double_t> Erf_pushforward(T x, T d_x)
+{
+   return {::TMath::Erf(x), (2 * ::TMath::Exp(-x * x) / ::TMath::Sqrt(::TMath::Pi())) * d_x}; // Doubt
+}
+
+template <typename T>
+ValueAndPushforward<Double_t, Double_t> Erfc_pushforward(T x, T d_x)
+{
+   return {::TMath::Erfc(x), -Erf_pushforward(x, d_x).pushforward};
+}
+
+template <typename T>
+ValueAndPushforward<Double_t, Double_t> Exp_pushforward(T x, T d_x)
+{
+   return {::TMath::Exp(x), ::TMath::Exp(x) * d_x};
+}
+
+template <typename T>
+ValueAndPushforward<T, T> Hypot_pushforward(T x, T y, T d_x, T d_y)
+{
+   return {::TMath::Hypot(x, y), x / ::TMath::Hypot(x, y) * d_x + y / ::TMath::Hypot(x, y) * d_y};
+}
+
+template <typename T, typename U>
+void Hypot_pullback(T x, T y, U p, clad::array_ref<T> d_x, clad::array_ref<T> d_y)
+{
+   T h = ::TMath::Hypot(x, y);
+   *d_x += x / h * p;
+   *d_y += y / h * p;
+}
+
+template <typename T>
+ValueAndPushforward<Double_t, Double_t> Log_pushforward(T x, T d_x)
+{
+   return {::TMath::Log(x), (1. / x) * d_x};
+}
+
+template <typename T>
+ValueAndPushforward<Double_t, Double_t> Log10_pushforward(T x, T d_x)
+{
+   return {::TMath::Log10(x), (1.0 / (x * ::TMath::Ln10())) * d_x};
 }
+
+template <typename T>
+ValueAndPushforward<Double_t, Double_t> Log2_pushforward(T x, T d_x)
+{
+   return {::TMath::Log2(x), (1.0 / (x * ::TMath::Log(2))) * d_x};
+}
+
+template <typename T>
+ValueAndPushforward<T, T> Max_pushforward(T x, T y, T d_x, T d_y)
+{
+   bool b = (x >= y);
+   return {::TMath::Max(x, y), (b ? 1 : 0) * d_x + (b ? 0 : 1) * d_y};
+}
+
+template <typename T, typename U>
+void Max_pullback(T a, T b, U p, clad::array_ref<T> d_a, clad::array_ref<T> d_b)
+{
+   if (a >= b)
+      *d_a += p;
+   else
+      *d_b += p;
+}
+
+template <typename T>
+ValueAndPushforward<T, T> Min_pushforward(T x, T y, T d_x, T d_y)
+{
+   bool b = (x <= y);
+   return {::TMath::Min(x, y), (b ? 1 : 0) * d_x + (b ? 0 : 1) * d_y};
+}
+
+template <typename T, typename U>
+void Min_pullback(T a, T b, U p, clad::array_ref<T> d_a, clad::array_ref<T> d_b)
+{
+   if (a <= b)
+      *d_a += p;
+   else
+      *d_b += p;
+}
+
+template <typename T>
+ValueAndPushforward<T, T> Power_pushforward(T x, T y, T d_x, T d_y)
+{
+   T pushforward = y * ::TMath::Power(x, y - 1) * d_x;
+   pushforward += (::TMath::Power(x, y) * ::TMath::Log(x)) * d_y;
```

Comment:
Here that would probably not work if the base is a negative number. You can see more discussion and how to address this in vgvassilev/clad#507

Reply:
Yes thank you for pointing it out. I have addressed it now and added a test case as well for it.